magda-web-client/src/Components/RequestDataset/RequestFormTemplate.js

```diff
@@ -180,6 +235,13 @@ export default class RequestFormTemplate extends React.Component {
                 >
                     {this.props.isSending ? "Sending..." : "Send"}
                 </AUbutton>
+                <AUbutton
+                    onClick={this.handleClear}
+                    className="au-btn--secondary correspondence-clear-button"
+                    disabled={this.state.clearButtonDisabled}
```

Comment:
hmm i'm not quite sure what you mean by this? you mean using react lifecycle methods?
I don't see how that will make this function like it is supposed to. This state is changed based on user entering and deleting into the text box.

Reply:
I mean rather than setting that in the state, you could just have something like
```javascript
...
disabled={!(this.state.message && this.state.message !== "") ||
      (this.state.senderName && this.state.senderName !== "") ||
      (this.state.senderEmail && this.state.senderEmail !== "")}
...
```